### ISSUE TITLE
feat: add Tokenlists resolver for tokens images

### ISSUE THREAD

**Issue by bonustrack** [snapshot-labs/stamp#24]
We should add a resolver for Tokenlists for tokens images, and use the Uniswap + Zerion + Coingecko lists for now see https://tokenlists.org/token-list?url=tokenlist.zerion.eth

**Comment by Sekhmet** [comment]
@bonustrack 
- [Zerion](https://tokenlists.org/token-list?url=tokenlist.zerion.eth) list doesn't seem to include images: https://wispy-bird-88a7.uniswap.workers.dev/?url=http://tokenlist.zerion.eth.link
- Coingecko and Uniswap has images, but they are super small (https://assets.coingecko.com/coins/images/9545/thumb/band-protocol.png?1568730326) I've found this variant where we can get better version at 50x50 (https://assets.coingecko.com/coins/images/9545/small/band-protocol.png?1568730326) - we would need to replace the URL from the tokenlist as the list has only 25x25 sizes. Tried to go bigger but no such links seem to work.

**Comment by samuveth** [comment]
You could also use this endpoint in combination with the contract addresses you get from the token lists https://www.covalenthq.com/docs/api/#/0/Get%20historical%20token%20prices/USD/1

**Comment by bonustrack** [comment]
@Sekhmet Good catch, let's forget put Tokenlists support on standby for now, I find another alternative using Zapper will create an issue for it. 

@samuveth I can't find an endpoint on Covalent get list of tokens directly (instead of a list of token based on some wallet balance).

**Comment by samuveth** [comment]
> You could also use this endpoint in combination with the contract addresses you get from the token lists https://www.covalenthq.com/docs/api/#/0/Get%20historical%20token%20prices/USD/1

@bonustrack you can use this endpoint, you will need to do one request for each token but it should be fine because we only need to do it once for all tokens.

**Comment by bonustrack** [comment]
It's annoying to have the list within the repository, it force to redeploy to see the updated data, create lot of commits that will be noise in changelog. I'd rather leverage the current caching system to store the data once it get too old.

**Comment by bonustrack** [comment]
My idea is to not use boost function, this create a loading time to start the server that is not needed, running it on the first reuquest is a more elegant approach. For the cache system we could store the date of latest cache within the cache and reset it if its larger than a day or so, this way we would avoid having to restart the server all the time, the ideal would be to not have to restart the server anytime.

**Comment by bonustrack** [comment]
@mktcode It's not for retry, it's to avoid resolving tokenlists multiple time if multiple request are sent at the same time.

**Comment by wa0x6e** [comment]
You're worrying about multiple concurrent requests triggering a promise to the same resource ?

We're using this to mitigate that https://github.com/snapshot-labs/snapshot-sequencer/blob/master/src/helpers/requestDeduplicator.ts

**Comment by wa0x6e** [comment]
Closing as not planned: token icons now come from trustwallet and DefiLlama (#641), which cover the gap this was meant to fill without syncing token lists. #282 targets a resolver layout that no longer exists.
